packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -10,22 +11,110 @@ import {
 import path from 'path'
 import { bundleWorkerEntry } from './worker'
 import { parseRequest } from '../utils'
-import { ENV_PUBLIC_PATH } from '../constants'
+import { ENV_ENTRY, ENV_PUBLIC_PATH } from '../constants'
 import MagicString from 'magic-string'
+import type { ViteDevServer } from '..'
+
+type WorkerType = 'classic' | 'module' | 'ignore'
 
 const WORKER_FILE_ID = 'worker_url_file'
 
+// find the workerOptions end with `)`
+function lexWorkerOptions(code: string, pos: number) {
+  let pattern = ''
+  const opStack = []
+
+  for (let i = pos; i < code.length; i++) {
+    const char = code.charAt(i)
+    if (char === '(') {
+      opStack.push(char)
+    } else if (char === ')') {
+      if (opStack.length) {
+        opStack.pop()
+      } else {
+        break
+      }
+    }
+    pattern += char
+  }
+
+  // had `,` split worker params
+  const commaIndex = pattern.indexOf(',')
+  if (commaIndex > -1) {
+    pattern = pattern.substring(commaIndex + 1)
+  }
+  return pattern
+}
```

Comment:
I wonder if we should re-use the utility from `importGlob.ts` instead?

https://github.com/vitejs/vite/blob/79428ad5b849455e14f95d1b439ae296ba231221/packages/vite/src/node/importGlob.ts#L200-L224

I think this function can be prone to comments with unclosed `(` and `)`